### ISSUE TITLE
AMD Text Draw Issues

### ISSUE THREAD

**Issue by Degeso** [sultim-t/xash-rt#3]
I'm aware that this wasn't tested with AMD, so I thought I'd share some insight on what's going on here!

The game itself launches and actually renders the world, seemingly with no issues (that I could see, given the text mess that you're about to wittness).

![image](https://user-images.githubusercontent.com/19645657/220761962-43024b19-40e3-428d-84c7-e8002409591b.png)

![image](https://user-images.githubusercontent.com/19645657/220762891-7c47c8fb-6332-4530-925c-7d408df079f0.png)

Sadly, this is what happens to... basically all texts and overlays shown over the rendered game world. That's in the main menu, and also when starting the game, when the black background fades away. In fact, as soon as the fading is done, the artifacts are gone too.

Card in question is a 7900XTX, so if I can assist with fixing issues, I'll happily help!

[engine.log](https://github.com/sultim-t/xash-rt/files/10808474/engine.log)

**Comment by magikmw** [comment]
I don't want to be the 'works for me' guy, but I've just tried it, and so far runs flawlessly on my RX 6600 XT.
Happy to compare notes.

**Comment by luisaugustz** [comment]
> I don't want to be the 'works for me' guy, but I've just tried it, and so far runs flawlessly on my RX 6600 XT. Happy to compare notes.

yeah, same, works for me on a 6600xt, using 22.1.2

maybe its tied to the 7000 series or a newer driver?

**Comment by Nopileus** [comment]
Can confirm this happens on 7900xt with driver 23.2.2
Appears to render and perform fine otherwise.

Edit: The same thing happens in vkquake-rt 1.0.2

**Comment by yoshinatsu** [comment]
> I don't want to be the 'works for me' guy, but I've just tried it, and so far runs flawlessly on my RX 6600 XT. Happy to compare notes.

Same. No such glitch for me on RX 6600 XT with 23.2.2. I was actually shocked it even works at all on AMD.

**Comment by funkkiy** [comment]
Doesn't run for me on RX 6600.

![image](https://user-images.githubusercontent.com/4707200/220814166-54bb600a-2df1-4890-8e57-123acdda6fa7.png)

**Comment by yoshinatsu** [comment]
> Doesn't run for me on RX 6600.
> 
> ![image](https://user-images.githubusercontent.com/4707200/220814166-54bb600a-2df1-4890-8e57-123acdda6fa7.png)

Are you using the Pro drivers by any chance?

**Comment by funkkiy** [comment]
No, but I am running 22.11.2 because I found it more stable than the latest drivers. Has anyone tested with this version before?

**Comment by yoshinatsu** [comment]
> No, but I am running 22.11.2 because I found it more stable than the latest drivers. Has anyone tested with this version before?

No need to use older drivers anymore.
Update to the latest ones, and make sure [you disable MPO](https://nvidia.custhelp.com/app/answers/detail/a_id/5157/~/after-updating-to-nvidia-game-ready-driver-461.09-or-newer%2C-some-desktop-apps). It's the culprit for 90% of RX 6000 issues.

**Comment by funkkiy** [comment]
Ah, I fixed my issue by reinstalling Half-Life! I disabled MPO too, but I had done that a few days ago to fix unrelated graphical issues.

**Comment by denliner** [comment]
I have no issue from my side (RX 6800XT with driver 23.2.1 with MPO disabled)
This seem to be an RDNA 3 issue.

**Comment by chaosophy** [comment]
I see the same issue as the OP.

6700XT with 23.2.2 drivers on WIn10.
Fresh install of Half-Life.

**Comment by Srandista** [comment]
Maybe @kd-11 can save us again?

**Comment by AnTrii** [comment]
Can confirm exactly same issue as OP with AMD drivers 23.2.1 and 23.2.2 on 7900 XTX, Win10 22h2 x64,

**Comment by thewacokid** [comment]
Same issue.  Affects all text overlay including the HUD.  7900XTX, 23.2.1 driver, Win11.

**Comment by thewacokid** [comment]
![image](https://user-images.githubusercontent.com/3431253/221075413-1304c2ea-0878-444d-8883-33098ca78e06.png)
It does appear to only be the bottom 1/3 of the screen on RDNA3.  Apologies for the Woot window. :)

**Comment by kd-11** [comment]
> Maybe @kd-11 can save us again?

I'm unavailable for the next few days unfortunately. I'll definitely look into it when I have some time (on monday)

**Comment by tomek0055** [comment]
I have also this issue on my 7900 XTX. Driver version is 23.2.2 on Win10 22h2

**Comment by kd-11** [comment]
I submitted a fix for this which solves the issue on my RDNA2 GPU. I don't have RDNA3 yet, so I don't know if the 7900 series has other issues.

**Comment by vvtunes** [comment]
7900XT here, have exactly the same problem (something to do with sprites?)
Was thinking it could due to a defective card, but other ray traced games seem to work fine.

(As a temporary workaround i have disabled all hud elements)

Happens in all engines (latest release): prboom+ rt, xash rt, vkquake rt and serious engine rt

Apart from this, everything is running smoothly, love the new half life lighting...

**Comment by chaosophy** [comment]
> I submitted a fix for this which solves the issue on my RDNA2 GPU. I don't have RDNA3 yet, so I don't know if the 7900 series has other issues.

Think I've done this correctly?

I downloaded the 'RayTracedGL1-Bundle' from here...

https://github.com/sultim-t/RayTracedGL1/actions/runs/4298133075

unzipped it and copied over what I think are the correct files to the correct locations and from a quick test can confirm it has fixed the issue, thanks!

**Comment by tomek0055** [comment]
> > I submitted a fix for this which solves the issue on my RDNA2 GPU. I don't have RDNA3 yet, so I don't know if the 7900 series has other issues.
> 
> Think I've done this correctly?
> 
> I downloaded the 'RayTracedGL1-Bundle' from here...
> 
> https://github.com/sultim-t/RayTracedGL1/actions/runs/4298133075
> 
> unzipped it and copied over what I think are the correct files to the correct locations and from a quick test can confirm it has fixed the issue, thanks!

I did the same, and I don't have artifacts anymore. Thanks for fix! 🙏

**Comment by Degeso** [comment]
> I submitted a fix for this which solves the issue on my RDNA2 GPU. I don't have RDNA3 yet, so I don't know if the 7900 series has other issues.

Can confirm that this fixes it on my 7900 XTX as well! 👍

**Comment by AnTrii** [comment]
Placing a new `RayTracedGL1.dll`  from [RayTracedGL1-Bundle.zip](https://github.com/sultim-t/RayTracedGL1/suites/11266011339/artifacts/577419230)\bin to `...\Steam\steamapps\common\Half-Life\rt_bin` folder definitely solves the issue with a text drawing!

**Comment by sultim-t** [comment]
Should be in [1.0.5a](https://github.com/sultim-t/xash-rt/releases/tag/1.0.5a)
